nixos/modules/system/boot/kernel.nix

```diff
@@ -36,6 +36,7 @@ in
 
     boot.kernelPackages = mkOption {
       default = pkgs.linuxPackages;
+      type = types.unspecified // { merge = mergeEqualOption; };
```

Comment:
I feel like this might give some odd error messages, because in order for Nix to be able to know whether two sets are equal, it needs to fully evaluate them, which could evaluate all packages, which could fail because some of them are broken or so. I haven't tested anything though.

Reply:
This did evaluate subpackages, which is why I had to conditionnally include some kernel modules such as e1000e. I left the asserts in case these files were imported somewhere else.
My config evaluates so I think it's fine.

Reply:
After this update an errors:
```
error: The option `boot.kernelPackages' has conflicting definitions, in `/home/user/works/src-nix/nixpkgs/nixos/modules/profiles/hardened.nix' and `/home/user/works/nixops/config-defs/core.nix'.
```
Use this configuration:
```
  boot = mkDefault {
    kernelPackages = pkgs.linuxPackages_5_2_hardened;
    loader.grub.enable = true;
    loader.grub.version = 2;
    cleanTmpDir = true;
    tmpOnTmpfs = false;
  };
```

Reply:
is that a nixops file or yours ? can't find it. Seems like it works as intended otherwise.

Reply:
My nixops configuration. Used for multiple remote systems.

Reply:
@joachifm @teto  really added this patch?
```
diff --git a/nixos/modules/profiles/hardened.nix b/nixos/modules/profiles/hardened.nix
index 626d8b1d2bd..2b00a015e3d 100644
--- a/nixos/modules/profiles/hardened.nix
+++ b/nixos/modules/profiles/hardened.nix
@@ -10,7 +10,7 @@ with lib;
     maintainers = [ maintainers.joachifm ];
   };

-  boot.kernelPackages = mkDefault pkgs.linuxPackages_hardened;
+  boot.kernelPackages = mkOverride 1500 pkgs.linuxPackages_hardened;

   nix.allowedUsers = mkDefault [ "@users" ];
```
With this patch, the nixops correct works.

Reply:
I didn't have any issue importing the hardened profile (nixos-usntable) on nixops (master).
Maybe if you provide a way a minimal example to reproduce it, I could look into it.

Reply:
@teto my example nixops config - https://elven.pw/temp/nixops_example.tar